classes/Kohana/Log.php

```diff
@@ -23,9 +23,24 @@ class Kohana_Log {
 	const DEBUG     = LOG_DEBUG;    // 7
 
 	/**
+	 * Numeric log level to string lookup table.
+	 * @var array
+	 */
+	protected $_log_levels = array(
+		LOG_EMERG   => \Psr\Log\LogLevel::EMERGENCY,
+		LOG_ALERT   => \Psr\Log\LogLevel::ALERT,
+		LOG_CRIT    => \Psr\Log\LogLevel::CRITICAL,
+		LOG_ERR     => \Psr\Log\LogLevel::ERROR,
+		LOG_WARNING => \Psr\Log\LogLevel::WARNING,
+		LOG_NOTICE  => \Psr\Log\LogLevel::NOTICE,
+		LOG_INFO    => \Psr\Log\LogLevel::INFO,
+		LOG_DEBUG   => \Psr\Log\LogLevel::DEBUG,
+	);
+
+	/**
 	 * @var  boolean  immediately write when logs are added
 	 */
-	public static $write_on_add = FALSE;
+	protected $write_immediately = FALSE;
```

Comment:
Every protected instead of private variable is a variable that can only in future be renamed/refactored in a breaking version release. For package code, we should probably lean to private by default unless there's a good reason to make it more visible. After all, if someone has a problem we can easily make it `private` => `protected` in a minor version release. We can't go the other way.